packages/@aws-cdk/aws-ec2/lib/client-vpn.ts

```diff
@@ -0,0 +1,409 @@
+import logs = require('@aws-cdk/aws-logs');
+import cdk = require('@aws-cdk/core');
+import {
+  CfnClientVpnAuthorizationRule,
+  CfnClientVpnEndpoint,
+  CfnClientVpnRoute,
+  CfnClientVpnTargetNetworkAssociation
+} from './ec2.generated';
+import {CIDR_VALIDATION_REGEXES} from "./peer";
+import {ISubnet} from './vpc';
+
+interface IClientAuthenticationRequestOptions {
+  /**
+   * Information about the Active Directory to be used
+   */
+  readonly activeDirectory?: {
```

Comment:
This isn't going to work in jsii (and if it does, it shouldn't).

Either inline the `directoryId` into the parent struct, or define it as an `IActiveDirectory` with a class that implements it.